src/schema/v2/vanityURLEntity.ts

```diff
@@ -0,0 +1,61 @@
+import { PartnerType } from "./partner"
+import { FairType } from "./fair"
+import {
+  GraphQLFieldConfig,
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { ResolverContext } from "types/graphql"
+
+export const VanityURLEntityType = new GraphQLUnionType({
+  name: "VanityURLEntityType",
+  types: [FairType, PartnerType],
+  resolveType(value, _context, _info) {
+    switch (value.context_type) {
+      case "Partner":
+        return PartnerType
+      case "Fair":
+        return FairType
+      default:
+        throw new Error(`Unknown context type: ${value.context_type}`)
+    }
+  },
+})
+
+const VanityURLEntity: GraphQLFieldConfig<void, ResolverContext> = {
+  type: VanityURLEntityType,
+  description: "A Partner or Fair",
+  args: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The slug or ID of the Profile to get a partner or fair for",
+    },
+  },
+  resolve: (_root, { id }, { profileLoader, partnerLoader, fairLoader }) =>
+    profileLoader(id).then((profile) => {
+      const ownerType = profile.owner?.type ?? profile.owner_type // what is the difference between these fields?
+      switch (ownerType) {
+        case "Gallery":
+        case "PartnerGallery":
+        case "Institution":
+        case "PartnerInstitution":
+        case "PartnerInstitutionalSeller":
```

Comment:
I think these should be any subclass of `Partner` (and maybe `Partner` itself?)

The full list seems to be:

 - `PartnerArtistDocument`
 - `PartnerAuction`
 - `PartnerBrand`
 - `PartnerDemo`
 - `PartnerGallery`
 - `PartnerInstitution`
 - `PartnerInstitutionalSeller`
 - `PartnerPrivateCollector`
 - `PartnerPrivateDealer`

Although at the very minimum we should cover the ones actually in use

![image](https://user-images.githubusercontent.com/1242537/90769111-9eeee300-e2e7-11ea-8c55-587fd7dd7333.png)

which seems to be all except `PartnerArtistDocument`.

Reply:
🙇‍♂️ That is a nifty trick I need to up my ruby/rails fu, was wondering how to find a canonical list of these. I'll update the list.